ui/src/dashboards/components/DashboardHeader.tsx

```diff
@@ -197,6 +136,60 @@ class DashboardHeader extends Component<Props> {
       onSetAutoRefreshStatus(dashboard.id, AutoRefreshStatus.Active)
     }
   }
+
+  return (
+    <>
+      <Page.Header fullWidth={true}>
+        <RenamablePageTitle
+          maxLength={DASHBOARD_NAME_MAX_LENGTH}
+          onRename={handleRenameDashboard}
+          name={dashboard && dashboard.name}
+          placeholder={DEFAULT_DASHBOARD_NAME}
+        />
+      </Page.Header>
+      <Page.ControlBar fullWidth={true}>
+        <Page.ControlBarLeft>
+          <Button
+            icon={IconFont.AddCell}
+            color={ComponentColor.Primary}
+            onClick={handleAddCell}
+            text="Add Cell"
+            titleText="Add cell to dashboard"
+          />
+          <Button
+            icon={IconFont.TextBlock}
+            text="Add Note"
+            onClick={handleAddNote}
+          />
+          <Button
+            icon={IconFont.Cube}
+            text="Variables"
+            onClick={toggleShowVariablesControls}
+            color={
+              showVariablesControls
+                ? ComponentColor.Secondary
+                : ComponentColor.Default
+            }
+          />
+          <DashboardLightModeToggle />
+          <PresentationModeToggle />
+          <GraphTips />
+        </Page.ControlBarLeft>
+        <Page.ControlBarRight>
+          <TimeZoneDropdown />
+          <AutoRefreshDropdown
+            onChoose={handleChooseAutoRefresh}
+            onManualRefresh={onManualRefresh}
+            selected={autoRefresh}
+          />
+          <TimeRangeDropdown
+            onSetTimeRange={handleChooseTimeRange}
+            timeRange={timeRange}
+          />
+        </Page.ControlBarRight>
+      </Page.ControlBar>
+    </>
+  )
```

Comment:
it's hard to tell in this diff, did any code change here, or is this just a reshuffling of code?

Reply:
just added 
```  
useEffect(() => {
    fireDashboardViewedEvent(dashboard.name)
  }, [dashboard.id])
```